### PR TITLE
feat: Combobox MVP [MDS-339]

### PR DESCRIPTION
**Pull request by kinowashere**
Combobox MVP, no multi included
Screenshot 2023-02-09 at 14 12 39
Screenshot 2023-02-09 at 15 42 37

## Checklist

- [x] You attached screenshots or added description about changes
- [x] You use [conventional commits naming](https://www.conventionalcommits.org/en/v1.0.0/), e.g. `fix: your changes description [TicketNumber]`, `feat: your feature name [TicketNumber]`
- [x] You have updated the documentation accordingly.

### CODE REVIEW

File: workspaces/core/src/combobox/Combobox.tsx [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import React, { Fragment, ReactNode, useState } from 'react';
+import { Combobox as HUICombobox, Transition } from '@headlessui/react';
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:2]: I think we need to replace the abbreviation HUI :-) as it is not really a polite world in Russian

File: workspaces/core/src/combobox/Combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                style={styles}
+                {...attributes?.popper}
+                className={mergeClassnames(
+                  'w-full min-w-[18.75rem] max-h-[18.75rem] z-1 py-2 px-1 my-2 rounded-moon-s-md box-border bg-gohan shadow-moon-lg overflow-y-auto',
+                  'focus:outline-none'
+                )}
+              >
+                {hasRenderProp ? children?.({ open }) : children}
+              </HUICombobox.Options>
+            </Transition>
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx]: this is better to move to subComponent Options, to have possibilities for customization with className

File: workspaces/core/src/combobox/Combobox.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+                  size === 'xl' && 'top-4 right-4',
+                  size == 'md' && 'top-10 right-2',
+                  size === 'lg' && 'top-11 right-2',
+                  size === 'sm' && 'top-10 right-2',
+                  open && 'rotate-[-180deg]'
+                )}
+              >
+                <ControlsChevronDownSmall />
+              </HUICombobox.Button>
+            </div>
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:114]: this is better to move to subComponent Input, to have possibilities for customization with className

File: workspaces/core/src/combobox/Combobox.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+                  size === 'sm' ? 'text-moon-16' : 'text-moon-24',
+                  size === 'xl' && 'top-4 right-4',
+                  size == 'md' && 'top-10 right-2',
+                  size === 'lg' && 'top-11 right-2',
+                  size === 'sm' && 'top-10 right-2',
+                  open && 'rotate-[-180deg]'
+                )}
+              >
+                <ControlsChevronDownSmall />
+              </HUICombobox.Button>
```
Comment (yarema184) [inline workspaces/core/src/combobox/Combobox.tsx:113]: need to check how this component looks in RTL mode

**Review by yarema184 — CHANGES_REQUESTED** [review]

File: workspaces/core/src/combobox/Combobox.tsx (hunk 1 above)
Comment (beygi) [reply workspaces/core/src/combobox/Combobox.tsx]: fixed

File: workspaces/core/src/combobox/Combobox.tsx (hunk 4 above)
Comment (beygi) [reply workspaces/core/src/combobox/Combobox.tsx]: Looks pretty well now, added the screenshot

**Review by yarema184 — APPROVED** [review]

**Pull request merged**